activemount.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+type activeMount struct {
+	UsageCount int
+}
+
+// activateVolume checks if the volume that has been requested to be mounted (as in docker volume mounting)
+// actually requires to be mounted (as an overlay fs mount). For that purpose check if other containers
+// have already mounted the volume (by reading in `activemountsdir`). It is also possible that the volume
+// has already been been mounted by the same container (when doing a `docker cp` while the container is running),
+// in that case the file named with the `requestId` will contain the number of times the container has
+// been requested the volume to be mounted. That number will be increased each time `activateVolume` is
+// called and decreased on `deactivateVolume`.
+// Parameters:
+//
+//	requestName: Name of the volume to be mounted
+//	requestID: ID of the container requesting the mount
+//	activemountsdir: Folder where Docker-On-Top mounts are tracked.
+//
+// Return:
+//
+//	doMountFs: Returns true if the request requires the filesystem to be mounted, false if not.
+//	err: If the function encountered an error, the error itself, nil if everything went right.
```

Comment:
```suggestion
//	doMountFs: `true` if the caller shall mount the filesystem, `false` otherwise.
//		If an error is returned, `doMountFs` is always `false`.
//	err: An error, if encountered, `nil` otherwise.
```

Reply:
Done